File: LinkedList/02_LinkedList/Operations.java
class Operations{
    static class Node{
        int data;
        Node next;

        Node(int data){
            this.data = data;
            this.next = null;
        }
    }

    // Will display the all nodes in a linked list.
    public static void display(Node node){
        Node temp = node;
        while(temp != null){
            System.out.println(temp.data);
            temp = temp.next;
        }
        System.out.println();
    }

    // Delete any node in the linkedlist
    public static Node deleteNode(Node node, int value){
        Node prev = null;
        Node curr = node;

        if(node!= null && node.data == value){
            return node.next;
        }

        while(curr != null){
            if(curr.data == value){
                if(prev != null){
                    prev.next = curr.next;
                    break;
                } 
            }
            prev = curr;
            curr = curr.next;
        }

        return node;
    }


    //Find length of a linkedlist
    public static int Length(Node node){
        Node temp = node;
        int count = 0;
        while(temp != null){
            count++;
            temp = temp.next;
        }

        return count;
    }

    public static boolean checkIfPresent(Node node, int ele){
        Node temp = node;
        while(temp != null){
            if( temp.data == ele){
                return true;
            }
            temp = temp.next;
        }

        return false;
    }
    public static void main(String[] args) {
        Node node = new Node(1);
        node.next = new Node(2);
        node.next.next = new Node(3);
        node.next.next.next = new Node(4);

        System.out.println("----Displaying all nodes in the LinkedList----");
        display(node);
        System.out.println("Length of the linkedlist is: "+ Length(node));
        System.out.println(checkIfPresent(node, 3));
        node = deleteNode(node, 3);
        System.out.println("----Displaying all nodes after deleting a node from the LinkedList----");
        display(node);
        System.out.println("Length of the linkedlist after deletion: "+ Length(node));
        System.out.println(checkIfPresent(node, 3));
    }
}

File: LinkedList/02_LinkedList/Operations.cpp
#include<iostream>
using namespace std;

typedef struct Node{
    int data;
    struct Node* next;

    Node(int data1){
        data = data1;
    }
} Node;


void display(Node* node){
    Node* temp = node;
    while(temp!= nullptr){
        cout<<temp->data<<endl;
        temp=temp->next;
    }
    cout<<endl;
}

Node* deleteNode(Node* node, int value){
    Node* prev = nullptr;
    Node* curr = node;
    if(curr!= nullptr && curr->data == value){
        Node* temp = curr->next;
        delete curr;
        return temp;
    }
    
    while(curr != nullptr){
        if(curr->data == value){
            prev->next = curr->next;
            delete curr;
            break;
        }

        prev = curr;
        curr = curr->next;
    }

    return node;
}

int length(Node* node){
    Node* temp = node;
    int count = 0;

    while(temp != nullptr){
        temp = temp->next;
        count++;
    }

    return count;
}

bool checkIfPresent(Node* node, int value){
    Node* temp = node;
    while(temp != nullptr){
        if(temp->data == value){
            return true;
        }
        temp = temp -> next;
    }

    return false;
}

void freeList(Node* node){
    Node* temp;
    while(node != nullptr){
        temp = node;
        node = node->next;
        delete temp;
    }
}

int main(){
    Node* node = new Node(1);
    node->next = new Node(2);
    node->next->next = new Node(3);
    node->next->next->next = new Node(4);

    cout<<"----Displaying the all nodes of linklist----"<<endl;
    display(node);
    node = deleteNode(node, 3);
    cout<<"----Displaying the all nodes of linklist after delete----"<<endl;
    display(node);
    cout<<"Length of the linkedList-"<<length(node)<<endl;
    cout<<checkIfPresent(node,4)<<endl;
    return 0;
}

